api/proto/teleport/crownjewel/v1/crownjewel.proto

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.crownjewel.v1;
+
+import "google/protobuf/wrappers.proto";
+import "teleport/header/v1/resourceheader.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/crownjewel/v1;crownjewelv1";
+
+// CrownJewel represents a Crown Jewel resource.
+// Crown Jewel is a resource that represents a set of resources that are
+// considered critical to the organization. Access Graph uses Crown Jewel to
+// generate audit events if access to a resource has changed.
+message CrownJewel {
+  // ResourceHeader is the resource header.
+  teleport.header.v1.ResourceHeader header = 1;
+  // Spec is the crown jewel spec.
+  CrownJewelSpec spec = 2;
+}
+
+// CrownJewelSpec is the specification of a Crown Jewel.
+message CrownJewelSpec {
+  // TeleportMatchers is a list of teleport matchers.
+  repeated TeleportMatcher teleport_matchers = 1;
+  // AWSMatchers is a list of AWS matchers.
+  repeated AWSMatcher aws_matchers = 2;
+}
+
+// TeleportMatcher represents a matcher for Teleport resources.
+message TeleportMatcher {
+  // Name is the name of the resource. When the name is provided, it will match
+  // the resource with the same name.
+  string name = 1;
+  // Kind is the kind of the resource: ssh, k8s, db, etc
+  // Multiple kinds can be provided to match multiple kinds.
+  repeated string kinds = 2;
+  // Labels is a set of labels.
+  map<string, string> labels = 3;
```

Comment:
Use the same approach as aws and just enforce it to be non-nil

Reply:
@tigrato I can, but in almost all places in our internal implementation, we're using maps to store our labels. Even our metadata struct uses it https://github.com/gravitational/teleport/blob/5bc69e06654ff4eea89b658326bfd2cf2bbb6aed/api/proto/teleport/header/v1/metadata.proto#L33
so I'm not sure why here we should diverge from it?

Reply:
In reality we should allow:

`key:[]*string`  where the string works as an or operator. 

That's how we did all matchers in teleport because you might want to select env: [prod, super-prod] without having to duplicate the matchers.

My idea is to

repeated TagMatcher labels = 5;


message TAGMatcher {
    string Key = 1;
    repeated google.protobuf.StringValue values =2;
}